pkg/db/advisory_detail.go

```diff
@@ -77,3 +79,65 @@ func (dbc Config) saveAdvisories(tx *bolt.Tx, bkt *bolt.Bucket, bktNames []strin
 func (dbc Config) DeleteAdvisoryDetailBucket() error {
 	return dbc.deleteBucket(advisoryDetailBucket)
 }
+
+// SaveRhsaAdvisoryDetails Extract 'RHSA' advisories from 'advisory-detail'-'Red Hat' bucket and copy them in each
+func (dbc Config) SaveRhsaAdvisoryDetails(tx *bolt.Tx, vulnID string, rejectedCve []string) error {
+	root := tx.Bucket([]byte(advisoryDetailBucket))
+	if root == nil {
+		return nil
+	}
+
+	cveBucket := root.Bucket([]byte(vulnID))
+	if cveBucket == nil {
+		return nil
+	}
+
+	redHatBucket := cveBucket.Bucket([]byte("Red Hat"))
+	if redHatBucket == nil {
+		return nil
+	}
+
+	if err := dbc.saveRhsaAdvisories(tx, redHatBucket, []string{"Red Hat"}, vulnID, rejectedCve); err != nil {
+		return xerrors.Errorf("walk advisories error: %w", err)
+	}
+
+	return nil
+}
```

Comment:
What is this function for...?

Reply:
this function finds all `Red Hat` buckets, decodes them, removes rejected cves and saves to db as ` SaveAdvisoryDetails`.